src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/PreferAsSpanOverSubstringTests.cs

```diff
@@ -0,0 +1,1854 @@
+        {
+            get
+            {
+                string receiver = CS.Usings + @"
+public class R
+{
+    public static void Consume(string a, string b, string c) { }
+    public static void Consume(Roschar a, string b, string c) { }
+    public static void Consume(string a, string b, Roschar c) { }
+    public static void Consume(Roschar a, Roschar b, string c) { }
+}";
+                yield return new[]
+                {
+                    receiver,
+                    VB.WithBody(WithKey(@"R.Consume(foo.Substring(1), foo.Substring(2), foo.Substring(3))", 0)),
+                    VB.WithBody(@"R.Consume(foo.AsSpan(1), foo.AsSpan(2), foo.Substring(3))")
+                };
+
+                receiver = CS.Usings + @"
+public class R
+{
+    public static void Consume(int n, string b, string c) { }
+    public static void Consume(double n, Roschar b, Roschar c) { }
+    public static void Consume(int n, string b, Roschar c) { }
+}";
+                yield return new[]
+                {
+                    receiver,
+                    VB.WithBody(WithKey(@"R.Consume(7, foo.Substring(2), foo.Substring(3))", 0)),
+                    VB.WithBody(@"R.Consume(7, foo.Substring(2), foo.AsSpan(3))")
+                };
+            }
+        }
+
+        [Theory]
+        [MemberData(nameof(Data_MultipleCandidateOVerloads_SingleBestCandidate_VB))]
+        public Task MultipleCandidateOverloads_SingleBestCandidate_ReportedAndFixed_VB(string receiverClass, string testCode, string fixedCode)
+        {
+            var project = new ProjectState("ReceiverProject", LanguageNames.CSharp, "receiver", "cs")
+            {
+                Sources = { receiverClass }
+            };
+
+            var test = new VerifyVB.Test
+            {
+                TestState =
+                {
+                    Sources = { testCode },
+                    AdditionalProjects = { { project.Name, project } },
+                    AdditionalProjectReferences = { project.Name },
+                    ExpectedDiagnostics = { VB.DiagnosticAt(0) }
+                },
+                FixedState =
+                {
+                    Sources = { fixedCode },
+                    AdditionalProjects = { { project.Name, project } },
+                    AdditionalProjectReferences = { project.Name }
+                },
+                ReferenceAssemblies = ReferenceAssemblies.Net.Net50
+            };
+            return test.RunAsync();
+        }
+
+        public static IEnumerable<object[]> Data_MultipleCandidateOverloads_Ambiguous_CS
+        {
+            get
+            {
+                string members = @"
+public void Consume(string a, string b) { }
+public void Consume(Roschar a, string b) { }
+public void Consume(string a, Roschar b) { }";
+                yield return new[]
+                {
+                    CS.WithBody(WithKey(@"Consume(foo.Substring(1), foo.Substring(2))", 0) + ';', members)
```

Comment:
We can potentially consider showing 2 code fixes here, one for each best candidate, but given this would be a rare case, current approach of not showing code fix is also reasonable.

Reply:
@mavasani I'm happy to implement that functionality. How do I write tests that expect two code fixes for one violation? Is there another analyzer that does this that I could look at as an example?

Reply:
@sharwell Might have suggestions on this - I don't believe we need this as part of this PR. Feel free to file a separate issue or open a follow-up PR for it.